pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -691,9 +691,10 @@ File getSegmentDataDir(String segmentName, @Nullable String segmentTier, TableCo
       File tierTableDataDir = new File(tierDataDir, _tableNameWithType);
       return new File(tierTableDataDir, segmentName);
     } catch (Exception e) {
-      LOGGER.warn("Failed to get dataDir for segment: {} of table: {} on tier: {} due to error: {}", segmentName,
-          _tableNameWithType, segmentTier, e.getMessage());
-      return getSegmentDataDir(segmentName);
+      File defaultDir = getSegmentDataDir(segmentName);
+      LOGGER.debug("Failed to get dataDir for segment: {} of table: {} on tier: {} due to error: {}. Use default: {}",
```

Comment:
Is this a common exception? If this is unexpected, we shouldn't log in debug level

Reply:
This is exception for TierBasedSegmentDirectoryLoader, which expects tier is backed by a data path and it logs a warn log itself. But it's a common case that a table has multi tiers backed by different server pools, which simply use the default dataDir to load segments via DefaultSegmentDirectoryLoader, i.e. no particular data path for the tiers. And because this is BaseTableDataManager class, so make this a debug log.